File: src/main/java/mods/eln/sim/process/heater/ResistorHeatThermalLoad.java
package mods.eln.sim.process.heater;

import mods.eln.sim.IProcess;
import mods.eln.sim.ThermalLoad;
import mods.eln.sim.mna.component.Resistor;

public class ResistorHeatThermalLoad implements IProcess {

    Resistor resistor;
    ThermalLoad load;

    public ResistorHeatThermalLoad(Resistor r, ThermalLoad load) {
        this.resistor = r;
        this.load = load;
    }

    @Override
    public void process(double time) {
        load.movePowerTo(resistor.getPower());
    }
}

File: src/main/java/mods/eln/sim/process/heater/ResistorHeatThermalLoad.kt
package mods.eln.sim.process.heater

import mods.eln.sim.core.IProcess
import mods.eln.sim.thermal.ThermalLoad
import mods.eln.sim.mna.passive.Resistor

class ResistorHeatThermalLoad(internal var r: Resistor, internal var load: ThermalLoad) : IProcess {

    override fun process(time: Double) {
        load.movePowerTo(r.getPower())
    }
}
